### PR TITLE
Resolve custom sample deploy error.

### LINKED ISSUE #944: Add "name" field in custom's yaml, To solve the deploy Error
/kind bug
Deploy the custom sample, got the error
```
error: error validating "xxx.yaml": error validating data: ValidationError(InferenceService.spec.default.predictor.custom.container): missing required field "name" in org.kubeflow.serving.v1alpha2.InferenceService.spec.default.predictor.custom.container; if you choose to ignore these errors, turn validation off with --validate=false
```

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
kubectl apply -f hellow_world/custom.yaml
**What did you expect to happen:**
Deploy the service succeed.
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:1.3.4
- Knative Version:0.14.0
- KFServing Version:master branch
- Kubernetes version: (use `kubectl version`):1.15.3
- OS (e.g. from `/etc/os-release`):ubuntu16

### PR DESCRIPTION
**Pull request by Iamlovingit**
Fixes #944
Add container name in custom sample's yaml file
Relax custom container name, remove the default custom container name
Remove the container name comparison.

### CODE REVIEW

**Comment by ellistarn** [comment]
/lgtm
I'll let Dan approve.

**Comment by Iamlovingit** [comment]
> /lgtm
> I'll let Dan approve.

thanks @ellistarn

File: pkg/apis/serving/v1alpha2/framework_custom.go [hunk 1]
```diff
@@ -55,8 +54,5 @@ func (c *CustomSpec) Validate(config *InferenceServicesConfig) error {
 	if err != nil {
 		return fmt.Errorf("Custom container validation error: %s", err.Error())
 	}
-	if c.GetStorageUri() != "" && c.Container.Name != constants.InferenceServiceContainerName {
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha2/framework_custom.go:58]: actually this validation is really for the purpose of storage initializer injection, if the model server container name is not `InferenceServiceContainerName` the storage initializer won't get injected for custom if user intend to use it. https://github.com/kubeflow/kfserving/blob/dafd2e6997a6decd4457af62896c72f025e8c2a5/pkg/webhook/admission/pod/storage_initializer_injector.go#L96

**Pull request merged**